### PR TITLE
Expose guide to adding a new edition as rustdoc

### PR DESCRIPTION
**Pull request by weihanglo**
### What does this PR try to resolve?

It's useful if the guide to adding a new edition can be exposed via rustdoc. Also add more intra doclinks for `features` module doc.

### How should we test and review this PR?

Run `cargo doc --document-private-items --no-deps --open` and read.

### CODE REVIEW

File: src/cargo/core/features.rs [hunk 1]
```diff
@@ -112,7 +115,45 @@ pub const SEE_CHANNELS: &str =
...
+/// The following sections will guide you how to add and stabilize an edition.
+///
+/// ## Adding a new edition
+///
+/// - Add the next edition to the enum.
+/// - Update every match expression that now fails to compile.
+/// - Update the [`FromStr`] impl.
+/// - Update [`CLI_VALUES`] to include the new edition.
+/// - Set [`LATEST_UNSTABLE`] to Some with the new edition.
+/// - Add an unstable feature to the `features!` macro invocation below for the new edition.
```
Comment (epage) [inline src/cargo/core/features.rs:129]: intradoc link for `features!`?

File: src/cargo/core/features.rs [hunk 2]
```diff
@@ -112,7 +115,45 @@ pub const SEE_CHANNELS: &str =
...
+/// - Gate on that new feature in TomlManifest::to_real_manifest.
```
Comment (epage) [inline src/cargo/core/features.rs:130]: Intradoc link?

File: src/cargo/core/features.rs [hunk 3]
```diff
@@ -112,7 +115,45 @@ pub const SEE_CHANNELS: &str =
...
+/// - Update the shell completion files.
+/// - Update any failing tests (hopefully there are very few).
+/// - Update unstable.md to add a new section for this new edition (see [this example]).
+///
+/// ## Stabilization instructions
+///
+/// - Set [`LATEST_UNSTABLE`] to None.
+/// - Set [`LATEST_STABLE`] to the new version.
+/// - Update [`is_stable`] to `true`.
+/// - Set the editionNNNN feature to stable in the `features!` macro invocation below.
```
Comment (epage) [inline src/cargo/core/features.rs:140]: intradoc link?

**Review by epage — COMMENTED** [review]
Thanks!

Just noticed a couple more intra-doc links that could be added

File: src/cargo/core/features.rs (hunk 1 above)
Comment (weihanglo) [reply src/cargo/core/features.rs]: Looks like it's a bit tricky to deal with the scope of macros for intra-doc links (https://github.com/rust-lang/rust/issues/72243). I hardcodes URLs then.

**Review by epage — APPROVED** [review]

**Pull request merged**